internal/logger/logger.go

```diff
@@ -42,6 +42,14 @@ var (
 	defaultLogger        *slog.Logger
 )
 
+// Logger exports the slog logger.
+// This allows one to use the full breadth of features provided by slog.
+//
+// TODO: export defaultLogger instead of using this function.
+func Logger() *slog.Logger {
```

Comment:
I think we should rename this function to 'SLogger' or something more explicit to make the caller code clearer.